### DISCUSSION TITLE
MavSDK vs MavLinkCom

### DISCUSSION THREAD

**Discussion by tank104** [microsoft/AirSim#4736] [General]
Hi,
I am just getting to grips with AirSim after moving from Gazebo.

Just wanted to confirm is MavLinkCom a replacement of sorts for MavSDK?  I can't find anything that confirms this or compares them, so was curious in the first instance if this was the case.

If so - and with AirSim discontinued, should I keep with MavSDK?  Is there a compelling reason to use MavLinkCom?

Thanks,

Andrew

**Comment by xxEoD2242** [comment]
Hey @tank104 ,

From my understanding, MavLinkCom is a custom-built C++ library written by the AirSim authors to communicate specifically with PX4 and with questionable integration with ArduPilot. This was done with the prospect of being able to provide tight coupling with AirSim. It doesn't replace the MavSDK C++ library, which you should probably be using for continuity sake, since Microsoft shut this project down.

However, my company has taken over the open source project and rebranded this to be Colosseum. See here -> https://github.com/CodexLabsLLC/Colosseum

We are working on improvements to the PX4/Ardupilot integration and will most likely move to MavSDK C++ library, as it is better supported across the Mavlink family of vehicles.

Let me know if you have questions on either platform.

**Comment by tank104** [comment]
Thanks for quick response!  I think I will leave MavLinkCom then as you mention.
I looked at Collosseum before - for now it looks very similiiar, but I guess only very recently branched.  
I will definitely keep on it - thanks!
